module/VuFind/src/VuFind/Search/Base/Params.php

```diff
@@ -1797,6 +1807,16 @@ public function deminify($minified)
         $this->query = QueryAdapter::deminify($minified->t);
     }
 
+    /**
+     * Get optional parameters.
+     *
+     * @return array
+     */
+    public function getOptionalParameters(): array
```

Comment:
Thanks, @LuomaJuha, this solution looks very safe -- I'm confident it will do what is desired without causing unanticipated side effects.

My only concern is that I don't really like the naming of "optional parameters" since it's not at all clear what that means -- it's very broad.

My suggestion is that perhaps we should rename them to "remembered pagination parameters" or something like that, and then we can also improve our comments to offer better clarity. For example:

```
/**
 * Get remembered pagination parameters from saved search. We track these separately since
 * in some contexts we want to use them (e.g. linking back to a search in breadcrumbs), but in
 * other contexts we want to ignore them (e.g. comparing two searches to see if they represent
 * the same query -- because page 1 and page 2 still represent the same overall search).
 *
 * @return array
 */
public function getRememberedPaginationParameters(): array
```

Maybe that's too verbose, and we could use "getLastPaginationParameters" or "getSavedPaginationParameters". Or maybe that's too narrow, and we need a broader term than "pagination" to encompass other things that might be saved (like view mode, for example). I'm definitely open to other ideas -- but hopefully you see the general direction I'm trying to move this in. :-)

Reply:
Yea, that is true! Hmm, I'll see if my brain comes up with any reasonable name, and if not I'll stick with this plan.

Reply:
I guess we could also pick @EreMaijala's brain for opinions if needed. :-)

Reply:
getSearchControlParams is something that my brain was thinking. Don't know if it sounds good enough

Reply:
How about getSearchContextParameters? "Control" suggests to me that the parameters could change the way the search behaves, but Context is perhaps more suggestive of where we are within the search.

Reply:
Well, we now have more room for longer function names! :)

Reply:
True!